src/internal/channel.js

```diff
@@ -124,16 +124,27 @@ export function eventChannel(subscribe, buffer = buffers.none(), matcher) {
   }
 
   const chan = channel(buffer)
+  const close = () => {
+    if(!chan.__closed__) {
+      if (unsubscribe) {
+        unsubscribe()
+      }
+      chan.close()
```

Comment:
ive debugged it to the moment ive realised that this was caused by this ordering `unsubscribing` vs `chan.close` and I think its reasonable anyway to have this fixed ordering of things, it seems... better, dont you think? i couldnt find any argument against this